packages/mason_logger/lib/src/mason_logger.dart

```diff
@@ -223,11 +223,11 @@ class Logger {
       final byte = _stdin.readByteSync();
       if (event.length == 3) event.clear();
       event.add(byte);
-      if (_upKey.every(event.contains) || _jKey.every(event.contains)) {
```

Comment:
This pattern has appeared twice for checking how the index should be updated.

Perhaps we could change how this is written to reduce repeated code.

Reply:
```dart
bool _eventIsOneOf(Iterable<int> event, Iterable<List<int>> keys) =>
  keys.any((key) => key.every(event.contains));
```
And use it like
```dart
if (_eventIsOneOf(event, [_upKey, _kKey])) {
  // handle up action
}
```

Reply:
Actually it's _6/7 times_ after taking a closer look

Reply:
yeah that sounds like an awesome improvement 👍

Reply:
You can even do it as an extension on Iterable<int> like:

```dart
extension on Iterable<int> {
  bool isOneOf(Iterable<int> items) {
    return items.any((item) => item.every(contains);
  }
}
```

And use like:

```dart
if (event.isOneOf([_upKey, _kKey])) {...}
```

Reply:
Yes, much easier to read 🙂